pkg/mount/mount.go

```diff
@@ -56,10 +57,17 @@ func Mounted(mountpoint string) (bool, error) {
 		return false, err
 	}
 
-	mountpoint, err = fileutils.ReadSymlinkedDirectory(mountpoint)
+	info, err := os.Stat(mountpoint)
 	if err != nil {
 		return false, err
 	}
+	if info.IsDir() {
+		mountpoint, err = fileutils.ReadSymlinkedDirectory(mountpoint)
+		if err != nil {
+			return false, err
+		}
+	}
+
```

Comment:
If it is NOT a dir, should we `return false, nil` in an else statement?

Reply:
No, then we just should look if the file is in the mounts :)

Reply:
@saschagrunert I'll trust your judgement on this, but I thought all mountpoints had to be directories in Linux?  New learning for me?

Reply:
Yeah for pinned linux namespaces in CRI-O we use one file mount per namespace. The issue was that the previous implementation would error out for the pinned namespaces, which cause  stale mounts.